File: javabackend/src/main/java/mft/dev/javabackend/dto/insert/TodoInsertRequestDTO.java
package mft.dev.javabackend.dto.insert;

import com.fasterxml.jackson.annotation.JsonProperty;
import mft.dev.javabackend.model.Todo;
import mft.dev.javabackend.model.Utente;

public record TodoInsertRequestDTO(
        @JsonProperty(value = "titolo", required = true)
        String titolo,
        @JsonProperty("descrizione")
        String descrizione
) {
    public Todo toTodo(Utente utente){
        Todo todo = new Todo();
        todo.setTitolo(this.titolo());
        todo.setDescrizione(this.descrizione());
        todo.setUtente(utente);

        return todo;
    }
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/dto/insert/TodoInsertRequestDTO.kt
package mft.dev.kotlinbackend.dto.insert

import com.fasterxml.jackson.annotation.JsonProperty
import mft.dev.kotlinbackend.model.Todo
import mft.dev.kotlinbackend.model.Utente

data class TodoInsertRequestDTO(
    @JsonProperty("titolo", required = true)
    val titolo: String,
    @JsonProperty("descrizione")
    val descrizione: String? = null
) {
    companion object{

    }

    fun toTodo(utente: Utente): Todo{
        return Todo(titolo = titolo, descrizione = descrizione, utente = utente)
    }
}
